app/components/GravityForm/Form.jsx

```diff
@@ -51,6 +51,8 @@ class GravityForm extends Component {
     const { title, description, button, fields, confirmation } = activeForm;
     // Submit failed watcher
     const { submitFailed } = this.state;
+    const currentForm = this.props.gravityforms[this.props.formId];
+    const postFailed = currentForm && currentForm.hasSubmitted && !currentForm.submitting && !currentForm.submitSuccess;
```

Comment:
f we have a current form and it has submitted, we then ensure that we are not submitting as a failed submittion previously that is currently being re-submitted is ok, then if we have a form, its submitted atleast once it's not longer submitting AND it wasn't successful, then post failed